Create DB and add migrations

AyoubM490 opened the pull request:
In this commit, I created the database and models for:
- users (with devise, and added columns such as a role for the authorization and
name
- motorcycles
- reservations
And each model has its migrations that define at the end of the schema.
I ran rails db:migrate on the app already.
 [x] closes https://github.com/sentayhu19/Book-an-appointment-back-end/issues/16

AyoubM490 committed:
Create DB and add migrations
In this commit, I created the database and models for:
- users (with devise, and added columns such as role for the authorization and
name
- motorcycles
- reservations
And each model has its migrations that define at the end the schema.
I runned rails db:migrate on the app already.

AyoubM490 committed:
fix linter errors

sentayhu19 commented on db/schema.rb:
```diff
@@ -0,0 +1,56 @@
+    t.integer "duration_months"
+    t.decimal "apr_rep"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
```
Good job on adding on these columns, but we also need 
`image `of the motorcycle
`description `of the motorcycle
`user_id` index reference

sentayhu19 commented on app/models/user.rb:
```diff
@@ -0,0 +1,6 @@
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+end
```
Good job on creating models, but we need also has_many relations here
`has_many :motorcycle` 🏍️

sentayhu19 commented on app/models/motorcycle.rb:
```diff
@@ -0,0 +1,2 @@
+class Motorcycle < ApplicationRecord
+end
```
Good job on creating models,but we need also belongs to relation to  a user
 `belongs_to :user`

sentayhu19 reviewed (changes requested):
Hi @AyoubMs 

Wonderful job so far! 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

![](https://media.giphy.com/media/tKyrc0NTlEcGWdThp4/giphy.gif)

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Achilles-Dev commented on db/migrate/20220811225830_devise_create_users.rb:
```diff
@@ -0,0 +1,44 @@
+  def change
+    create_table :users do |t|
+      ## Database authenticatable
+      t.string :email,              null: false, default: ""
+      t.string :encrypted_password, null: false, default: ""
```
Since we are only using only `username` field to authenticate a user, I would suggest you comment out the e-mail and password fields since they will not be required. Thank you.

Achilles-Dev commented on db/migrate/20220811231507_add_name_to_user.rb:
```diff
@@ -0,0 +1,5 @@
+class AddNameToUser < ActiveRecord::Migration[7.0]
+  def change
+    add_column :users, :name, :string
+  end
```
I suggest you make the name field a unique index for each user since that would be the only field for user authentication as stated in the requirement. Thank you

Achilles-Dev reviewed (changes requested):
Hi @AyoubMs 

Good job so far! 👏 
There are some issues that you still need to work on before you move on but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

AyoubM490 committed:
- remove email and password fields from devise model
- redo the motorcycle migration, with adding image, description and user_id
index reference
- add has_many motorcycles association to the user
- add belongs_to user association to the motorcycle
- add uniqueness validation of the name of the user model

AyoubM490 committed:
add name presence validation

AyoubM490 committed:
added name as a unique index

Achilles-Dev reviewed (approved):
## Hi @AyoubMs, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

![Image](https://media.giphy.com/media/qlrBlSDevEdFeW5JwV/giphy.gif)

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.
